sdk/python/kfp/dsl/__init__.py

```diff
@@ -43,9 +43,11 @@
 ]
 
 from kfp.components.component_decorator import component
+from kfp.components.container_component_decorator import container_component
 from kfp.components.importer_node import importer
 from kfp.components.pipeline_context import pipeline
 from kfp.components.pipeline_task import PipelineTask
+from kfp.components.structures import ContainerSpec
```

Comment:
As with `container_component`, `ContainerSpec` will now be included in our API reference docs once this branch is merged with master.

Can you refresh the docstrings of `ContainerSpec` for this? See the docstrings of [PipelineTaskFinalStatus](https://github.com/kubeflow/pipelines/blob/b10440d6a8a3a67f0b6d8ddeed6d8befbb98cdf5/sdk/python/kfp/components/task_final_status.py#L20-L55) for an example of how to document dataclasses in a way that works well with our documentation generator.

A few things I noticed are:
* put attribute docstrings below attributes
* remove the "The" prefix on all descriptions, capitalize first letter of first word of description
* remove `(optional)` (type annotations convey this already)
* perhaps state that this is object is only meant to be used by pipeline authors when constructing an `@container_component`